manifests/repos.pp

```diff
@@ -22,19 +34,66 @@
       $package_key   = $docker::package_key
       $package_repos = $docker::package_repos
 
+      if ( $facts['os']['distro']['id'] == 'Debian' and versioncmp($facts['os']['distro']['release']['major'],'10') >= 0 ) or ( $facts['os']['distro']['id'] == 'Ubuntu' and versioncmp($facts['os']['distro']['release']['major'],'22') >= 0 ) { # lint:ignore:140chars
```

Comment:
Debian 10 is the oldest supported OS, so you don't need to check for the version. And please use the `os.name` fact instead of `os.distro.id`. That requires the lsb tools and isn't always present.

Reply:
resolved with 6c53719936e4c560c9c0b37ace2233d426d288aa
I changed the versioncmp to verify debian version is atleast version 11, since debian 10 does not mark apt-key as deprecated.